data_processing/meteo/task_functions.py

```diff
@@ -29,44 +30,80 @@
 DATADIR = f"{AIRFLOW_DAG_TMP}meteo/data"
 with open(f"{AIRFLOW_DAG_HOME}{DAG_FOLDER}meteo/config/dgv.json") as fp:
     config = json.load(fp)
+hooks = ["latest", "previous"]
 
 
-def list_ftp_files_recursive(ftp, path='', base_path=''):
+def clean_hooks(string, hooks=hooks):
+    _ = string
+    for h in hooks:
+        _ = _.replace(f"{h}-", "")
+    return _
+
+
+def build_file_id(file, path):
+    # we are creating ids for files based on their name so that we can
+    # match files that have been updated with a name change
+    # for instance:
+    # QUOT_SIM2_latest-2020-202310.csv.gz becomes QUOT_SIM2_latest-2020-202311.csv.gz
+    # when Nov2023 is added, but it actually *is* just a content update of the same file
+    file_id = file
+    if config.get(path, {}).get("source_pattern"):
+        params = re.match(config[path]["source_pattern"], file)
+        params = params.groupdict() if params else {}
+        if "PERIOD" in params and any([
+            h in params["PERIOD"] for h in hooks
+        ]):
+            # this will have to change if more hooks are added
+            params["PERIOD"] = "latest" if "latest" in params["PERIOD"] else "previous"
```

Comment:
If we're mostly replacing `PERIOD`, we could use a regex to replace the period matched?